src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/ReviewTimeExtensionsHandler.java

```diff
@@ -41,31 +56,85 @@ public boolean canHandle(
+
+
         State currentState = callback.getCaseDetails().getState();
         List<IdValue<TimeExtension>> timeExtensions = getTimeExtensions(asylumCase).map(timeExtension -> {
             TimeExtension timeExtensionValue = timeExtension.getValue();
             if (currentState == timeExtensionValue.getState() && timeExtensionValue.getStatus() == SUBMITTED) {
 
                 TimeExtensionDecision timeExtensionDecision = getTimeExtensionDecision(asylumCase);
                 TimeExtensionStatus timeExtensionStatus = timeExtensionDecision == TimeExtensionDecision.REFUSED ? REFUSED : GRANTED;
-
+                if (timeExtensionStatus.equals(GRANTED)) {
+                    isGranted.set(true);
+                }
                 return new IdValue<>(timeExtension.getId(), new TimeExtension(
-                        timeExtensionValue.getRequestedDate(),
-                        timeExtensionValue.getReason(),
-                        timeExtensionValue.getState(),
-                        timeExtensionStatus,
-                        timeExtensionValue.getEvidence(),
-                        timeExtensionDecision,
-                        getTimeExtensionDecisionReason(asylumCase)
+                    timeExtensionValue.getRequestDate(),
+                    timeExtensionValue.getReason(),
+                    timeExtensionValue.getState(),
+                    timeExtensionStatus,
+                    timeExtensionValue.getEvidence(),
+                    timeExtensionDecision,
+                    getTimeExtensionDecisionReason(asylumCase),
+                    decisionOutcomeDueDate
                 ));
             }
             return timeExtension;
         }).collect(Collectors.toList());
 
+        Optional<List<IdValue<Direction>>> maybeDirections = asylumCase.read(DIRECTIONS);
+        Optional<List<IdValue<TimeExtension>>> maybeTimeExtensions = asylumCase.read(TIME_EXTENSIONS);
+
+        Optional<IdValue<Direction>> directionBeingUpdated = maybeDirections.orElse(emptyList())
+            .stream()
+            .filter(directionIdVale -> {
+                return directionIdVale.getValue().getTag().equals(DirectionTag.REQUEST_REASONS_FOR_APPEAL); //todo this needs to work for a number of states
+            }).findFirst();
+
+        List<IdValue<Direction>> changedDirections =
+            maybeDirections.orElse(emptyList())
+                .stream()
+                .map(idValue -> {
+                    if (String.valueOf(maybeDirections.orElse(emptyList()).size()).equals(idValue.getId())) {
+                        return new IdValue<>(
+                            idValue.getId(),
+                            new Direction(
+                                idValue.getValue().getExplanation(),
+                                idValue.getValue().getParties(),
+                                decisionOutcomeDueDate,
+                                dateProvider.now().toString(),
```

Comment:
Hum do we update the sent date on a direction when we change the due date can you check with Gareth please.